internal/declarative/v2/renderer_raw.go

```diff
@@ -2,31 +2,101 @@ package v2
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"github.com/kyma-project/lifecycle-manager/pkg/types"
+	"io"
 	"os"
 
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
 	"k8s.io/client-go/tools/record"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultBufferSize                                          = 2048
+	ConditionTypeRawManifestCRDs               ConditionType   = "RawManifestCRDs"
+	ConditionReasonRawManifestCRDsAreAvailable ConditionReason = "RawManifestCRDsAvailable"
 )
 
 func NewRawRenderer(
 	spec *Spec,
+	clt Client,
 	options *Options,
 ) Renderer {
 	return &RawRenderer{
 		EventRecorder: options.EventRecorder,
+		Client:        clt,
 		Path:          spec.Path,
 	}
 }
 
 type RawRenderer struct {
 	record.EventRecorder
+	Client
 	Path string
+	crds []*unstructured.Unstructured
+}
+
+func (r *RawRenderer) prerequisiteCondition(object metav1.Object) metav1.Condition {
+	return metav1.Condition{
+		Type:               string(ConditionTypeRawManifestCRDs),
+		Reason:             string(ConditionReasonRawManifestCRDsAreAvailable),
+		Status:             metav1.ConditionFalse,
+		Message:            "CustomResourceDefinitions from raw manifest resources are installed and ready for use",
+		ObservedGeneration: object.GetGeneration(),
+	}
 }
 
-func (r *RawRenderer) Initialize(_ Object) error {
+func (r *RawRenderer) Initialize(obj Object) error {
+	status := obj.GetStatus()
+
+	prerequisiteExists := meta.FindStatusCondition(status.Conditions, string(ConditionTypeRawManifestCRDs)) != nil
+	if !prerequisiteExists {
+		meta.SetStatusCondition(&status.Conditions, r.prerequisiteCondition(obj))
+		obj.SetStatus(status)
+		return ErrConditionsNotYetRegistered
+	}
+
 	return nil
 }
 
-func (r *RawRenderer) EnsurePrerequisites(_ context.Context, _ Object) error {
+func (r *RawRenderer) EnsurePrerequisites(ctx context.Context, obj Object) error {
```

Comment:
If it's a raw manifest, all the resources are already saved in that yaml, why we need to implement this ensure prerequisites? This file don't need to change.

Reply:
Yes. I agree in principle. But I faced problems related to missing CRDs when applying the raw manifest without ensuring that CRDs included in it are already installed. I couldn't find a different approach to that which fixes the CRD issue.

Reply:
Please always keep in mind we are using reconcile pattern, if some resources fail in the first several try, but it can deploy successfully eventually, then it's okay. You don't have to make sure CRD get deployed first, raw manifest anyway include CRD, it's just in this reconcile round, CRD is not being fully deployed, but the manifest controller anyway always tries to deploy/check resources consistencies in each reconcile loop, eventually all resources can be deployed.